docs/pages/technical-specs/expo-updates-0.md

```diff
@@ -0,0 +1,187 @@
+  runtimeVersion: string
+  launchAsset: Asset
+  assets: Asset[]
+  updateMetadata: { [key: string]: number | string }
+}
+```
+Where an _Asset_ is the JSON:
+```
+{
+  hash: string
+  key: string
+  contentType: string
+  url: string
+}
+```
+  * `id` The ID MUST uniquely specify the manifest.
+  * `createdAt` The date and time created is essential as the client-library selects the most recent update (subject to any constraints supplied by the `expo-manifest-filters` header). The datetime should be formatted according to [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601)
+  * `runtimeVersion` Can be any string defined by the developer. It stipulates what native code setup is required to run the associated JavaScript update bundle.
+  * `Asset` Provides information about an asset and where to obtain it.
+    * `hash` SHA256 hash of the file to guarantee integrity.
+    * `key` Key generated by the bundler and used to identify this asset.
+    * `contentType` The mime type of the asset as defined by [RFC 2045](https://tools.ietf.org/html/rfc2045). e.g. `application/javascript`, `image/jpeg`.
+    * `url` Location where the asset is hosted.
+  * `launchAsset` The asset that is used to launch the app.
+  * `assets` An array of assets used by the update bundle, such as JavaScript, pictures, and fonts.
+  * `updateMetadata` The metadata associated with an update. This can be used for filtering the update (see the `branchname` example above) and also for the creation of more helpful errors. In particular, the server MAY send back the string valued `updateMetadata` keys:
+    * `branchName` used to sort updates into groups called _branches_.
+    * `updateGroup` used to group updates for different platforms that come from the same publish.
+
+### Errors
+
+A conformant server SHOULD return 
+  * a `404` error if a manifest is not found.
+  * a `400` if the request is malformed.
+  * a `405` if the request is not a GET.
+  * a `500` if there is a server error.
+
+## Asset Request
+A conformant client-library MUST make a GET request to the asset URL specified by the manifest. The client-library SHOULD include a header accepting the asset's content type as specified in the manifest. Additionaly the client-library SHOULD specify the compression encoding the client-library is capable of handling.
+
+Example headers:
+```
+accept: image/jpeg
+accept-encoding: br, gzip
+```
+
+## Asset Response
+
+An asset hosted at a particular URL MUST NOT be changed. Client-libraries with stale updates may still require old assets.
+
+### Headers
+
+The asset MUST be served with an asset compressed in compliance with the client's encoding request. The asset MUST include a content-type header set to the mime type of the asset defined in the manifest.
+For example:
+```
+content-encoding: br
+content-type: application/javascript
+```
+
+The asset endpoints MUST be served with a `cache-control` header set to an appropriately short period of time. For example:
```

Comment:
Actually, why is this the case? If assets are immutable as specified above shouldn't the cache control be a long duration?

Reply:
This might have been copied over from the manifest endpoint section above, where it's ideal for manifests to be fresh.

As for immutability, you're right about that. For now, I wonder if it's best to say:
```suggestion
An asset is RECOMMENDED to be served with a `cache-control` header set to a long duration as an asset located at a given URL must not change. For example:

cache-control: public, max-age=31536000, immutable
```

Reply:
Yes, great catch!